lib/AST/GenericSignatureBuilder.cpp

```diff
@@ -3743,13 +3743,11 @@ ResolvedType GenericSignatureBuilder::maybeResolveEquivalenceClass(
     if (!nestedPA)
       return ResolvedType::forUnresolved(baseEquivClass);
 
-    if (resolutionKind != ArchetypeResolutionKind::AlreadyKnown) {
```

Comment:
I'm confused. As far as I can tell this change just means that `updateNestedTypeForConformance` now gets called with `ArchetypeResolutionKind::AlreadyKnown` in addition to the other two kinds, but, I can't see a way in which that call actually does anything. 

https://github.com/apple/swift/blob/adfead025138474a5b4a6f3c1dee585c5654cc42/lib/AST/GenericSignatureBuilder.cpp#L2796-L2934

It seems like that function does three "interesting" things, but none of them happen with this kind:

- processes delayed requirements, but this is conditional on the kind being `CompleteWellFormed` https://github.com/apple/swift/blob/adfead025138474a5b4a6f3c1dee585c5654cc42/lib/AST/GenericSignatureBuilder.cpp#L2813
- creates a new archetype, but this is explicitly not meant to happen for `AlreadyKnown` https://github.com/apple/swift/blob/adfead025138474a5b4a6f3c1dee585c5654cc42/lib/AST/GenericSignatureBuilder.cpp#L2847-L2868
- does "more processing" to an archetype, conditional on `shouldUpdatePA`, which is only changed to `true` when creating a new archetype, i.e. it doesn't happen for `AlreadyKnown` https://github.com/apple/swift/blob/adfead025138474a5b4a6f3c1dee585c5654cc42/lib/AST/GenericSignatureBuilder.cpp#L2872

What am I missing?

Reply:
Same question here...

Reply:
You're both right and... somewhere I messed something up locally, because the test marked as fixed doesn't seem to be fixed. Very sorry :(